transformer_thermal_model/schemas/thermal_model/input_profile.py

```diff
@@ -72,6 +72,28 @@ def load_profile_array(self) -> np.typing.NDArray[np.float64]:
         """Require subclasses to define a load_profile property."""
         raise NotImplementedError("Subclasses must define a load_profile field or property.")
 
+    @classmethod
+    def to_datetime_array(cls, obj: pd.DatetimeIndex | list | tuple | np.ndarray) -> np.typing.NDArray:
```

Comment:
Shouldnt this be a private function of this class? 

```suggestion
    def _to_datetime_array(cls, obj: pd.DatetimeIndex | list | tuple | np.ndarray) -> np.typing.NDArray:
```